optuna/pruners/_hyperband.py

```diff
@@ -136,10 +141,33 @@ def objective(trial):
             See the details for :class:`~optuna.pruners.SuccessiveHalvingPruner`.
     """
 
+    @overload
+    def __init__(
+        self,
+        min_resource: int = 1,
+        max_resource: "Literal['auto']" = "auto",
+        reduction_factor: int = 3,
+        bootstrap_count: "Literal[0]" = 0,
+    ) -> None:
+        ...
+
+    @overload
+    def __init__(
+        self,
+        min_resource: int = 1,
+        # NOTE: We must provide a defualt value or else we cannot write the arguments in order.
+        # This value is not used; if the case a default is not provided the original init is called
+        # and max_resource is set to "auto".
+        max_resource: int = 100,
+        reduction_factor: int = 3,
+        bootstrap_count: int = 0,
+    ) -> None:
+        ...
```

Comment:
I'm sorry I'm not sure about the behavior of the type checker, but do we need these overloads? In principle the type `Union[Literal['auto'], int]` should be enough, isn't it?